src/org/mirah/jvm/mirrors/better_scope.mirah

```diff
@@ -79,6 +79,10 @@ class Locals
     end
     type
   end
+  
+  def has_local_type(name:String):boolean
+    @local_types.containsKey(name)
```

Comment:
This seems like it should be unnecessary. Is this because has_local uses `@declared_locals`, which has elements removed when they resolve as an error?
Maybe we could see if changing `has_local` to check the key set instead would be better.

I'm also not sure why Locals removes elements when they are in error. I find that a little surprising.